source/includes/fundamentals/code-snippets/logging.go

```diff
@@ -0,0 +1,170 @@
+	"log"
+	"os"
+	"sync"
+
+	"github.com/bombsimon/logrusr/v4"
+	"github.com/joho/godotenv"
+	"github.com/sirupsen/logrus"
+	easy "github.com/t-tomalak/logrus-easy-formatter"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func main() {
+	if err := godotenv.Load(); err != nil {
+		log.Println("No .env file found")
+	}
+
+	var uri string
+	if uri = os.Getenv("MONGODB_URI"); uri == "" {
+		log.Fatal("You must set your 'MONGODB_URI' environmental variable. See\n\t https://www.mongodb.com/docs/drivers/go/current/usage-examples/#environment-variable")
+	}
+	//standardLogging(uri)
+	//customLogging(uri)
+	thirdPartyLogging(uri)
+}
+
+func standardLogging(uri string) {
+	// start-standard-logger
+	loggerOptions := options.
+		Logger().
+		SetMaxDocumentLength(25).
+		SetComponentLevel(options.LogComponentCommand, options.LogLevelDebug)
+
+	clientOptions := options.
+		Client().
+		ApplyURI(uri).
+		SetLoggerOptions(loggerOptions)
+	// end-standard-logger
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatalf("error connecting to MongoDB: %v", err)
+	}
+
+	defer client.Disconnect(context.TODO())
+
+	// start-insert
+	coll := client.Database("testDB").Collection("testColl")
+	_, err = coll.InsertOne(context.TODO(), bson.D{{"item", "grapefruit"}})
+	// end-insert
+
+	if err != nil {
+		log.Fatalf("InsertOne failed: %v", err)
+	}
+}
+
+// start-customlogger-struct
+type CustomLogger struct {
+	io.Writer
+	mu sync.Mutex
+}
+// end-customlogger-struct
+
+// start-customlogger-funcs
+func (logger *CustomLogger) Info(level int, msg string, _ ...interface{}) {
+	logger.mu.Lock()
+	defer logger.mu.Unlock()
+	if level == 1 {
+		fmt.Fprintf(logger, "level: %d DEBUG, message: %s\n", level, msg)
+	} else {
+		fmt.Fprintf(logger, "level: %d INFO, message: %s\n", level, msg)
+	}
+}
+
+func (logger *CustomLogger) Error(err error, msg string, _ ...interface{}) {
+	logger.mu.Lock()
+	defer logger.mu.Unlock()
+	fmt.Fprintf(logger, "error: %v, message: %s\n", err, msg)
+}
+// end-customlogger-funcs
+
+func customLogging(uri string) {
+	// start-set-customlogger
+	buf := bytes.NewBuffer(nil)
+	sink := &CustomLogger{Writer: buf}
+
+	loggerOptions := options.
+		Logger().
+		SetSink(sink).
+		SetComponentLevel(options.LogComponentCommand, options.LogLevelDebug).
+		SetComponentLevel(options.LogComponentConnection, options.LogLevelDebug)
+
+	clientOptions := options.
+		Client().
+		ApplyURI(uri).
+		SetLoggerOptions(loggerOptions)
+    // end-set-customlogger
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatalf("error connecting to MongoDB: %v", err)
+	}
+
+	defer client.Disconnect(context.TODO())
+
+	coll := client.Database("testDB").Collection("testColl")
+	_, err = coll.InsertOne(context.TODO(), bson.D{{"item", "grapefruit"}})
+	if err != nil {
+		log.Fatalf("InsertOne failed: %v", err)
+	}
+	fmt.Println(buf.String())
```

Comment:
Should we use the log package to print the results from the custom log writer? 
```suggestion
	log.Println(buf.String())
```